json_encoder_bench_test.go

```diff
@@ -34,6 +34,18 @@ type logRecord struct {
 	Fields  map[string]interface{} `json:"fields"`
 }
 
+func BenchmarkJSONEncoderNest(b *testing.B) {
+	f := func(kv KeyValue) error {
+		kv.AddString("s", "string")
+		return nil
+	}
+	for i := 0; i < b.N; i++ {
+		enc := newJSONEncoder()
+		enc.Nest("nested", f)
+		enc.Free()
+	}
+}
```

Comment:
Does this benchmark reflect how we expect this method to be used? I'd expected most users to pass an anonymous function, like this:

```go
enc := newJSONEncoder()
enc.Nest("nested", func(kv KeyValue) error {
  kv.AddString("s", "string")
  return nil
})
enc.Free()
```